server/rpc.go

```diff
@@ -404,15 +440,27 @@ func SubmitSegment(bcast Broadcaster, seg *stream.HLSSegment, nonce uint64) (*ne
 		Seq:  int64(seg.SeqNo),
 		Hash: crypto.Keccak256(seg.Data),
 	}
+	uploaded := seg.Name != "" // hijack seg.Name to convey the uploaded URI
+
+	// send credentials for our own storage
+	if drivers.NodeStorage != nil && drivers.NodeStorage.IsExternal() {
+		segData.Storage = []*net.OSInfo{bcast.GetBroadcasterOS().GetInfo()}
```

Comment:
we're checking `drivers.NodeStorage`, but using `beast.GetBroadcasterOS()`, so this implicitly assumes that broadcaster's OS was created from `drivers.NodeStorage`

Reply:
Fixed and pushed but untested